src/dashboard/src/devine_dashboard/js/dashboard/canvas_drawer.js

```diff
@@ -0,0 +1,319 @@
+import { RosTopic } from "./ros";
+import { distinctColors } from "../vars/colors";
+import $ from "jquery";
+
+const BODY_PARTS = {
+  EYES: {
+    LEFT: 15,
+    RIGHT: 14
+  },
+  AGGR: [
+    [1, 2],
+    [1, 5],
+    [2, 3],
+    [3, 4],
+    [5, 6],
+    [6, 7],
+    [1, 8],
+    [8, 9],
+    [9, 10],
+    [1, 11],
+    [11, 12],
+    [12, 13],
+    [1, 0],
+    [0, 14],
+    [14, 16],
+    [0, 15],
+    [15, 17]
+  ]
+};
+
+export default class CanvasDrawer {
+  /**
+   * Class who draws over a selected canvas element.
+   * @param {dict} devineTopics - The list of ros topics.
+   * @param {string} canvasId - The html element canvas id.
+   * @param {func|undefined} updatePositionsCallback - Object found callback.
+   * @param {func|undefined} updateDelayLengthCallback - Current delay lenght callback.
+   */
+  constructor(devineTopics, canvasId, updatePositionsCallback=undefined, updateDelayLengthCallback=undefined) {
+    this.updatePositionsCallback = updatePositionsCallback;
+    this.updateDelayLengthCallback = updateDelayLengthCallback;
+
+    this.topics = {
+      image_raw: new RosTopic(devineTopics.compressed_image),
+      segmentation_image: new RosTopic(devineTopics.segmentation_image),
+      body_tracking_image: new RosTopic(devineTopics.body_tracking_image),
+      zone_detection_image: new RosTopic(devineTopics.zone_detection_image_out),
+      confidence: new RosTopic(devineTopics.objects_confidence),
+      segmentation: new RosTopic(devineTopics.objects),
+      object_position_2d: new RosTopic(devineTopics.guess_location_image),
+      object_position_3d: new RosTopic(devineTopics.guess_location_world),
+      body_tracking: new RosTopic(devineTopics.body_tracking)
+    };
+
+    this.canvas = this._getCanvas(canvasId);
+    this.canvasSize = { x: 640, y: 480 };
+    this._drawNoFeed();
+    
+    this.history = this._initHistoryBuffers();
+    this.changeImageSource('image_raw');
+
+    this._initConfidenceReset();
+  }
+
+  /** Draw current image on canvas */
+  draw() {
+    // TODO: const draw = throttle(function draw() {}, 100);
```

Comment:
@tihbe I've removed that for now, but is it necessary to keep it?

Reply:
Not anymore no, we're throttling openni directly

Reply:
Perfect